app/lib/request.rb

```diff
@@ -11,9 +11,10 @@ class Request
   def initialize(verb, url, **options)
     @verb    = verb
     @url     = Addressable::URI.parse(url).normalize
-    @options = options.merge(socket_class: Socket)
+    @options = options.merge(socket_class: Socket).merge(Rails.configuration.x.http_client_proxy)
     @headers = {}
 
+    raise Mastodon::HostValidationError, 'blocked access to darknet' if !Rails.configuration.x.access_to_darknet && /\.(onion|i2p)$/.match(@url.host)
```

Comment:
- Use capital B at the beginning of message
- Extract the if-clause to a method

Reply:
"Instance does not support hidden service connections" would maybe be a better message for admins?

Reply:
You can use `@url.tld` btw